src/MongoDB.Driver.Core/Core/Authentication/GssapiAuthenticator.cs

```diff
@@ -218,15 +220,7 @@ public ISaslStep Initialize(IConnection connection, SaslConversation conversatio
 
                 if (_canonicalizeHostName)
                 {
-#if NETSTANDARD1_5 || NETSTANDARD1_6
-                    var entry = Dns.GetHostEntryAsync(hostName).GetAwaiter().GetResult();
-#else
-                    var entry = Dns.GetHostEntry(hostName);
-#endif
-                    if (entry != null)
-                    {
-                        hostName = entry.HostName;
-                    }
+                    hostName = _dnsResolver.GetHostNameWithReverseDnsLookup(hostName);
```

Comment:
As I see spec requires getting not simply canonical hostname value (CNAME), but CNAME with reversed DNS.
See this spec line: https://github.com/mongodb/specifications/commit/6cadd12ef9d0c04e1e1c7e7adf40cad2a3a4bc00#diff-4ae4e931060899b3147a7255caadd990R413

Reply:
![image](https://user-images.githubusercontent.com/44870738/69725586-8de5bc80-112f-11ea-9519-40e6675e4f6b.png)

Reply:
NOTE: the `CNAME` value for `LDAPTEST.10GEN.CC` is `ec2-54-225-237-121.compute-1.amazonaws.com`.
cc: @jyemin, @rstam

Reply:
Acknowledged